lib/src/react16_suggestors/react_style_maps_updater.dart

```diff
@@ -0,0 +1,299 @@
+  List affectedValues = const [],
+  bool isAVariable = false,
+  bool styleMapContainsAVariable = false,
+  bool isAFunction = false,
+  bool addExtraLine = false,
+  bool isForCustomProps = false,
+}) {
+  String checkboxWithAffectedValues = '// [ ] Check this box upon manual '
+      'validation that this style map uses a valid value for the following '
+      'keys: ${affectedValues.join(', ')}.';
+
+  String variableCheckbox = '// [ ] Check this box upon manual validation '
+      'that this style map is receiving a value that is valid for the keys '
+      'that are simple string variables.';
+
+  String variableCheckboxWithAffectedValues = '// [ ] Check this box upon '
+      'manual validation that this style map is receiving a value that is '
+      'valid for the following keys: ${affectedValues.join(', ')}.';
+
+  String willBeRemovedSuffix = '//$willBeRemovedCommentSuffix';
+
+  String functionCheckbox = '// [ ] Check this box upon manual validation '
+      'that the method called to set the style prop does not return any '
+      'simple, unitless strings instead of nums.';
+
+  if (isForCustomProps) {
+    return '''
+    $variableCheckbox
+    $styleMapExample
+    $willBeRemovedSuffix
+    ''';
+  } else if ((isAVariable || styleMapContainsAVariable)) {
+    if (affectedValues.isNotEmpty) {
+      return '''
+      $variableCheckboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    } else {
+      return '''
+      $variableCheckbox
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    }
+  } else if (isAFunction) {
+    return '''
+       $functionCheckbox
+       $styleMapExample
+       $willBeRemovedSuffix
+       ''';
+  } else {
+    if (addExtraLine) {
+      return '''
+    
+      $checkboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    } else {
+      return '''
+      $checkboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    }
+  }
+}
+
+String cleanString(dynamic elementToClean) {
+  return elementToClean.toString().replaceAll("\'", "").replaceAll("\"", "");
+}
+
+bool nodeIsLikelyAnExpression(String node) =>
+    node.contains('?') && node.contains(':') || node.contains('??');
+
+bool isANumber(String node) => num.tryParse(node) != null;
+
+bool isAString(String node) => node.contains('"') || node.contains('\'');
+
+bool nodeIsLikelyAVariable(String node) => !isAString(node) && !isANumber(node);
+
+/// Removes the '...style' and '=' entities and returns the third entity.
+///
+/// The third entity will be the style map, a method invocation, or a variable.
+dynamic getStyles(Expression cascade) {
+  List styleCascadeEntities = cascade.childEntities.toList();
+  List cleanedChildEntities = List.from(styleCascadeEntities);
+
+  cleanedChildEntities.removeRange(0, 2);
+
+  return cleanedChildEntities.first;
+}
+
+bool hasValidationComment(AstNode node, SourceFile sourceFile) {
```

Comment:
Isn't this copy-pasted from somewhere else? We should put these functions in a common utility file so the implementations don't get out of sync

Reply:
Good call! I wasn't sure if that was something to do here or before the parent ticket merged, so I will do that now!